src/cloudai/workloads/megatron_run/megatron_run.py

```
@@ -80,9 +80,23 @@ def no_dashed_args(self):
 
     @property
     def cmd_args(self) -> dict[str, Union[str, list[str]]]:
+        # Fields with default="" are boolean flags; extra fields (model_extra) are also boolean by convention.
+        bool_flags = {k for k, f in MegatronRunCmdArgs.model_fields.items() if f.default == ""}
+        extra_keys = set(self.model_extra or {})
         args = self.model_dump(exclude_none=True, exclude={"docker_image_url", "run_script"})
-        args = {f"--{k.replace('_', '-')}": v for k, v in args.items()}
-        return args
+        result: dict[str, Union[str, list[str]]] = {}
```

Ruff (0.15.21)

[warning] Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)